catalog/async/catalog-async-data/src/test/groovy/org/codice/ddf/catalog/async/data/LazyProcessResourceImplTest.groovy

```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.async.data
+
+import ddf.catalog.resource.Resource
+import org.codice.ddf.catalog.async.data.impl.LazyProcessResourceImpl
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.util.function.Supplier
+
+import static org.codice.ddf.catalog.async.data.impl.ProcessResourceImpl.*
+
+class LazyProcessResourceImplTest extends Specification {
```

Comment:
I think some test cases are missing here for all of the public methods when the resource fails to be loaded.

Reply:
Added test cases.